cmd/chartsvc/handler.go

```diff
@@ -321,6 +321,44 @@ func listChartsWithFilters(w http.ResponseWriter, req *http.Request, params Para
 	response.NewDataResponse(cl).Write(w)
 }
 
+// findCharts returns the list of charts that matches the query param in any of these fields:
+//  - name
+//  - description
+//  - repository name
+//  - any keyword
+//  - any source
+//  - any maintainer name
+func findCharts(w http.ResponseWriter, req *http.Request, params Params) {
```

Comment:
one minor thing that I forgot to mention, it might make more sense to call this `searchCharts`?